docs/Project-Configuration/Best-Practices.md

```diff
@@ -23,6 +23,67 @@ The reasons to prefer subprocess.exec include:
 
 You can pass environment variables to subprocess.exec if you'd like to pass expansions. It's a good idea to avoid shell.exec as much as possible.
 
+
```

Comment:
This is great, thanks for adding it! 
One small suggestion: 
The section right above this ends with 

>  It's a good idea to avoid shell.exec as much as possible.

I think there's an opportunity to add a sentence or two to smooth over the transition. Maybe something like "While subprocess.exec is the safer option, if you have to use shell.exec, here are some tips on how to do it safely. Please test your setup with some test values to make sure that sensitive data doesn't get logged".

Reply:
Totally agreed with this -- I just pushed a new commit to address this comment, does that look OK to you?

Reply:
That looks perfect, thank you!